### DISCUSSION TITLE
Checkver: Access File with version from a private gitlab repository

### DISCUSSION THREAD

**Discussion by Ercolino** [ScoopInstaller/Scoop#5380] [Help and Questions] [answered]
Hi everyone! I have some problems with the checkver for my app. Background: the bucket as wells as the app are located into gitlab repos. To download the app executable I use a gitlab api endpoint and I added in scoop config the private host to make the download possible. So far I could locally added the bucket and downloaded the app. Now, I have problems by autoupdating. In the app repository there is a txt file where the latest version of the app has been written.  The checkver url points to this file and with a small regex should read the correct string. This seems to not be working.. I executed the checkver.ps1 script locally but nothing happens. It would be helpful to get more logs.. I wonder if it could be an authentication problem and if I could use some headers inside the checkver object

**Comment by rashil2000** [comment, marked as answer]
checkver does not read the private_hosts config variable. This is a bug. Could you make a PR? This code https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/bin/checkurls.ps1#L66 can be added to bin/checkver.ps1 after this line https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/bin/checkver.ps1#L224

**Reply by Ercolino** [reply]
Hi, thank you for your reply. I opened a PR: https://github.com/ScoopInstaller/Scoop/pull/5381
